Launcher/Launcher/src/Helpers/StartupHelper.cs

```diff
@@ -0,0 +1,31 @@
+using WSL;
+
+namespace Launcher.Helpers;
+
+public static class StartupHelper {
+    private static bool booted;
+
+    public static bool BootDistro() {
+        if (booted) return true;
+        try {
+            WslApiLoader.WslLaunchInteractive(
+                DistributionInfo.Name,
+                "sh -c \"exit\"", // Don't run anything, exit after we get a shell
+                false,
+                out var exitCode
+            );
+            if (exitCode != 0) {
+                goto fail;
+            }
+        } catch (WslApiException) {
+            goto fail;
+        }
```

Comment:
Can you log this exception message instead of a generic error message?

Reply:
As you can see here [1], the Exception only contains the return code of the underlying WSL API function and no meaningful message. I have split the error handling now, so that it is clear, whether the command inside WSL or launching the command itself failed

[1]: https://github.com/nzbr/wsl.net/blob/main/src/WslApiException.cs